File: Erettsegi/emelt/2013_may_eng/Szamok.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <algorithm>
#include <set>

using namespace std;

struct Feladat{
	string question, category;
	int point, answer;
	Feladat(string quest, int ans, int poi, string cat){
		question = quest; answer = ans; point = poi; category = cat;
	}
};

int main(){
	vector<Feladat> feladatok;
	ifstream input("felszam.txt");
	string currentLine, cat;
	int ev, point;
	while(getline(input, currentLine)){
		input >> ev >> point >> cat;
		feladatok.emplace_back(currentLine, ev, point, cat);
		getline(input, currentLine);
	}
	input.close();
	
	cout << "Feladatok száma: " << feladatok.size() << endl;
	
	//[0]: matek, [1]: 1 pont, [2]: 2 pont, [3]: 3 pont, [4]: min, [5]: max
	int counters[] {0, 0, 0, 0, feladatok[0].answer, feladatok[0].answer};
	set<string> categories;
	
	for(Feladat &feladat : feladatok){
		categories.insert(feladat.category);
		if(feladat.point < counters[4]){
			counters[4] = feladat.point;
		}
		if(feladat.point > counters[5]){
			counters[5] = feladat.point;
		}
		if(feladat.category == "matematika"){
			++counters[0];
			if(feladat.point == 1){
				++counters[1];
			}else if(feladat.point == 2){
				++counters[2];
			}else{
				++counters[3];
			}
		}
	}
	
	cout << "Matek feladatok száma: " << counters[0] << ", 1 pontos feladatok: " << counters[1]
		 << ", 2 pontos feladatok: " << counters[2] << ", 3 pontosak: " << counters[3] << endl;
	
	cout << "Legkisebb feladat megoldás: " << counters[4] << ", legnagyobb: " << counters[5] << endl;
	
	for(string print : categories){
		cout << print << ' ';
	}
	cout << endl;
	
	string temakor;
	int valasz;
	cout << "Írj be 1 témakört!" << endl;
	cin >> temakor;
	
	Feladat fel = feladatok[rand() % feladatok.size()];
	for(; fel.category != temakor; fel = feladatok[rand() % feladatok.size()]);
	
	cout << fel.question << endl;
	cin >> valasz;
	
	if(valasz == fel.answer){
		cout << "Ügyi vagy! Itt 1 keksz és " << fel.answer << " pont!" << endl; 
	}else{
		cout << "Rossz válasz! 0 pont, helyes válasz: " << fel.answer << endl;
	}
	
	set<int> generalt;
	for(int r = rand() % feladatok.size(); generalt.size() != 10; r = rand() % feladatok.size()){
		generalt.insert(r);
	}
	
	ofstream output("tesztfel.txt");
	for(int kek : generalt){
		Feladat print = feladatok[kek];
		output << print.point << ' ' << print.answer << ' ' << print.question << endl;
	}
	output.close();
	return 0;
}

File: erettsegi/emelt/2013_may_eng/c#/Szamok.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;

var rand = new Random();
var feladatok = new List<Feladat>();
var lines = File.ReadAllLines("felszam.txt");

for(var k = 0; k < lines.Length - 1; k += 2) {
    var split = lines[k + 1].Split(' ');
    feladatok.Add(new Feladat(lines[k], int.Parse(split[0]), int.Parse(split[1]), split[2]));
}

Console.WriteLine($"{feladatok.Count} Feladat van a fájlban!");

var temakorok = new List<String>();
var matekCounters = new int[4];

foreach(var all in feladatok) {
    if(!temakorok.Contains(all.temakor)) {
        temakorok.Add(all.temakor);
    }

    if(all.temakor.Equals("matematika")) {
        ++matekCounters[0];

        if(all.pont == 1) {
            ++matekCounters[1];
        }else if(all.pont == 2) {
            ++matekCounters[2];
        }else {
            ++matekCounters[3];
        }
    }
}

Console.WriteLine($"Az adatfajlban {matekCounters[0]} matematika feladat van,\n1 pontot er: " +
                  $"{matekCounters[1]} feladat, 2 pontot er {matekCounters[2]} feladat, 3 pontot er {matekCounters[3]} feladat. ");

feladatok = feladatok.OrderBy(k => k.valasz).ToList();

Console.WriteLine($"A legkisebb válaszú feladat: {feladatok[0].valasz}, a legnagyobb: {feladatok[feladatok.Count - 1].valasz}");
Console.WriteLine($"Előforduló témakörök: {temakorok}");

Console.WriteLine("Írj be 1 témakört!");
var readCat = Console.ReadLine();
var categorizált = new List<Feladat>();

foreach(var all in feladatok) {
    if(all.temakor.Equals(readCat)) {
        categorizált.Add(all);
    }
}
var chosen = categorizált[rand.Next(categorizált.Count)];

Console.WriteLine(chosen.kerdes);

if(int.Parse(Console.ReadLine()) == chosen.valasz) {
    Console.WriteLine("Kapott pontszám: " + chosen.pont);
}else{
    Console.WriteLine("Rossz válasz, 0 pont...\nA helyes válasz: " + chosen.valasz);
}

var generalt = new List<Feladat>();
for(var k = 0; k < 10; ++k) {
    var randomFeladat = feladatok[rand.Next(feladatok.Count)];

    if(generalt.Contains(randomFeladat)) {
        --k;
    }else{
        generalt.Add(randomFeladat);
    }
}

var osszpont = 0;
using var output = new StreamWriter("tesztfel.txt");

foreach(var toPrint in generalt) {
    osszpont += toPrint.pont;
    output.WriteLine(toPrint.pont + " " + toPrint.kerdes);
}
output.Write(osszpont);